buildscripts/semantic-convention/templates/SemanticAttributes.scala.j2

```diff
@@ -0,0 +1,377 @@
+@nowarn("msg=never used")
+@nowarn("msg=message and version: @deprecated")
+object {{class}} {
+  /**
+   * The URL of the OpenTelemetry schema for these keys and values.
+   */
+  val SchemaUrl: String = "{{schemaUrl}}"
+  {%- for attribute in attributes if attribute.is_local and not attribute.ref %}
+
+  /**
+  * {{attribute.brief | render_markdown(code="{{@code {0}}}", paragraph="{0}")}}
+    {%- if attribute.note %}
+  *
+  * <p>Notes:
+    <ul> {{attribute.note | render_markdown(code="{{@code {0}}}", paragraph="<li>{0}</li>", list="{0}")}} </ul>
+    {%- endif %}
+    {%- if (attribute.stability | string())  == "StabilityLevel.DEPRECATED" %}
+  *
+  * @deprecated {{attribute.brief | to_doc_brief}}.
+    {%- endif %}
+  */
+    {%- if (attribute.stability | string()) == "StabilityLevel.DEPRECATED"  %}
+  @deprecated
+    {%- endif %}
+  val {{attribute.fqn | to_camelcase(True)}}: AttributeKey[{{upFirst(to_scala_return_type(attribute.attr_type | string))}}] = {{to_scala_key_type(attribute.attr_type | string)}}("{{attribute.fqn}}")
+  {%- endfor %}
+
+  // Enum definitions
+  {%- for attribute in attributes if attribute.is_local and not attribute.ref %}
+  {%- if attribute.is_enum %}
+  {%- set class_name = attribute.fqn | to_camelcase(True) ~ "Value" %}
+  {%- set type = to_scala_return_type(attribute.attr_type.enum_type) %}
+  sealed abstract class {{ class_name }}(value: {{ type }})
+  object {{class_name}} {
+    {%- for member in attribute.attr_type.members %}
+      /** {% filter escape %}{{member.brief | to_doc_brief}}.{% endfilter %} */
+      case object {{ member.member_id | to_camelcase(True) }} extends {{class_name}}({{ print_value(type, member.value) }})
+    {%- endfor %}
+
+    {%- if class_name == "NetTransportValue" %}
+    /** @deprecated This item has been removed as of 1.13.0 of the semantic conventions. */
+    @deprecated("This item has been removed", "1.13.0")
+    case object Ip extends NetTransportValue("ip")
+    /** @deprecated This item has been removed as of 1.13.0 of the semantic conventions. */
+    @deprecated("This item has been removed", "1.13.0")
+    case object Unix extends NetTransportValue("unix")
+    {%- endif %}
+
+  }
+
+  {% endif %}
+  {%- endfor %}
+
+  {%- if class == "SemanticAttributes" %}
+  // Manually defined and not YET in the YAML
+  /**
+   * The name of an event describing an exception.
+   *
+   * <p>Typically an event with that name should not be manually created. Instead {@link
+   * org.typelevel.otel4s.trace.Span#recordException(Throwable)} should be used.
+   */
+  val ExceptionEventName: String = "exception"
+
+  /**
+   * The name of the keyspace being accessed.
+   *
+   * @deprecated this item has been removed as of 1.8.0 of the semantic conventions. Please use {@link SemanticAttributes#DbName} instead.
+   */
+  @deprecated("Use SemanticAttributes.DbName instead", "1.8.0")
+  val DbCassandraKeyspace: AttributeKey[String] = string("db.cassandra.keyspace")
```

Comment:
Should we remove these deprecated ones?

Reply:
Based on the otel's specifications, we should keep them indefinitely :(
https://opentelemetry.io/docs/specs/otel/versioning-and-stability/#deprecated

Reply:
Hmm, do you mean this part?

> Signals SHALL only be marked as deprecated when the replacement becomes stable. Deprecated code MUST abide by the same support guarantees as stable code.

---

> we should keep them indefinitely

I agree that if we already published a release with them, then we should definitely keep them for backwards-compatibility (i.e. the same support guarantees as stable code). However, because we never released them when they were stable, it doesn't make sense to release them now as deprecated, right?

Reply:
I don't think it's solely about the SDKs. Rather, it's about the specification and its deprecations. I believe the keys should exist in any SDK to extract even the deprecated attributes in spans sent by old apps.

Reply:
I think the versions in the `@deprecated` tag ought to be the version of the library, not the version of the spec.  We can refer to versions of the spec in the descriptions.